### PR TITLE
Detect Steam installation dir on Windows

### PR DESCRIPTION
**Pull request by ivystopia**
The `--steamappspath` argument is now optional.

### CODE REVIEW

File: vgm-extractor.py [hunk 1]
```diff
@@ -55,16 +59,23 @@
...
+# Detect Steam installation directory (TODO: support non-windows OS)
+if args.steamappspath or not platform.startswith('win'):
+    if not Path(args.steamappspath).is_dir():
+        arg_parser.error(f'steamappspath is not a directory: {args.steamappspath}')
+else:
+    import winreg
+    try:
+        hkey = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, "SOFTWARE\WOW6432Node\Valve\Steam")
+        steam_install_path = winreg.QueryValueEx(hkey, "InstallPath")[0]
+        steamapps_path = Path(steam_install_path).joinpath("common")
```
Comment (sparr) [inline vgm-extractor.py]: Can you confirm it's InstallPath/common and not InstallPath/steamapps/common?

**Comment by ivystopia** [comment]
It should have been `steamapps/common`, updated

**Comment by sparr** [comment]
I can't test this right now. I like the improvement, and I can do the Linux equivalent, but merge here depends on someone else confirming that this works on windows.

**Comment by sparr** [comment]
I can't merge this with the source branch deleted, so I copied the code over and worked it in at [c2371a4](https://github.com/sparr/vgm-extractor/commit/c2371a4df414416885434085a92de9f5414c312d)

**Pull request closed without merging**